meiki_server/auth/auth.go

```diff
@@ -65,6 +70,18 @@ func CreateAuth(ctx context.Context, token_coll *mongo.Collection, user_coll *mo
 	return Auth{token_coll, user_coll}, nil
 }
 
+func validateCredentials(username string, password string) error {
+	if !usernameRegex.MatchString(username) {
+		return ErrInvalidUsername
+	}
+
+	if len(password) < 5 {
```

Comment:
probably a separate function at some point of time, but good enough for now

Reply:
Yep, I saw a lot of complex regex but would not prefer that, this is probably a seperate spike I guess